### ISSUE TITLE
[Select]: WCAG and other issues

### ISSUE THREAD

**Issue by ivarkallejarv** [TEHIK-EE/tedi-design-system#803]
## Problem
TEDI Version 11.3.3
All problems use TEDI-REDI select, unless specified otherwise

Functional
- Select loses focus, cannot navigate via keyboard, etc. (also described in https://github.com/TEHIK-EE/tedi-design-system/issues/743)
- ^ Related. Currently the select doesn't close upon clicking anywhere else as well, but probably related to focus loss

Visual + Accessibility
- No options help message has text color of rgb(153, 153, 153) in CSS, which only has a contrast of 2.85:1 with background of #fff 
![Image](https://github.com/user-attachments/assets/d5245eb3-bdda-46f3-8289-23307bfd8135)

- [Community Select] - Border color on hover has a contrast of 2.09:1
![Image](https://github.com/user-attachments/assets/3a197f37-6287-494d-a399-efd501f3861e)

- Width 1280px, zoom 400% - if you have more space on the top than on the bottom when opening select, it'll open in the upwards direction. And in such a case where the select has a lot of content, the content will be hidden behind the top border of the browser. Select content doesn't adjust to take available space into account
![Image](https://github.com/user-attachments/assets/1cfdf7a8-ed6c-4f8e-9e0c-0b56314cc192)

- Add content to renderMessageListFooter. Try to navigate to the footer content using tab navigation. Currently the Select closes and the content is not accessible
![Image](https://github.com/user-attachments/assets/971fa869-c72d-40db-825f-f83a93c2d2c5)

- Add content to renderMessageListFooter that is focusable. Focus the select input. Press tab. Focus disappears (probably trying to focus the buttons inside footer) and the select options closes, thus giving the illusion that the user's focus disappears entirely.
![Image](https://github.com/user-attachments/assets/01602472-50b5-43cb-b73d-5aa56517123e)

- Navigate inside the select, using JAWS. Options inside are not read out as clickable elements to the user
![Image](https://github.com/user-attachments/assets/6fec9bf9-a7cd-4964-b57e-e91efe9b4e3c)

- Add noOptionsMessage to Select. Navigate to Select using JAWS. JAWS doesn't read out the no options message to the user
![Image](https://github.com/user-attachments/assets/464a01ab-87c5-4177-8644-5df3e9ea2a0a)

## Examples and information

```
<Select
      value={value}
      onChange={(value) => onChange(value as ISelectOption<PersonSearchSelectCustomData> | null)}
      noOptionsMessage={() => t(`${isInputShort ? 'input-short' : 'no-result'}`)}
      loadingMessage={() => t('loading')}
      label={"Select label"}
      id="person-search"
      name={inputName}
      async={true}
      menuIsOpen={isMenuOpen}
      onMenuClose={() => setIsMenuOpen(false)}
      onMenuOpen={() => setIsMenuOpen(true)}
      onInputChange={handleInputChange}
      loadOptions={loadOptions}
      helper={helper}
      renderOption={(props) => (
        <Row gutterX={2}>
          {props.data.customData?.label && (
            <Col width="auto">
              <StatusBadge color={props.data.customData?.label === 'L' ? 'warning' : 'danger'}>
                {props.data.customData?.value}
              </StatusBadge>
            </Col>
          )}
          <Col>
            <Text color="brand">Some label</Text>
          </Col>
        </Row>
      )}
      renderMessageListFooter={() => (
        <VerticalSpacing size={0.75}>
          <Row justifyContent="center" gutterX={2}>
            <Col width="auto">
              <Button
                size="small"
                visualType="secondary"
                onTouchStart={() => doThis?.()}
                onClick={() => doThis?.()}
              >
                Do X
              </Button>
            </Col>
            <Col width="auto">
              <Button
                size="small"
                visualType="secondary"
                onTouchStart={() => doThat?.()}
                onClick={() => doThat?.()}
              >
                Do Y
              </Button>
            </Col>
          </Row>
          <Text modifiers={['small', 'center']} color="secondary">
            Info text here
          </Text>
        </VerticalSpacing>
      )}
      ref={innerRef}
    />
```

**Comment by tobre6** [comment]
1. Confirmed in TEDI-READY
2. Cannot reproduce
3. Confirmed in TEDI-READY
4. Problem in Community select. OK in TEDI-READY
5. Confirmed in TEDI-READY. Problem is 320x256px with 100% as well
6. Confirmed in TEDI-READY
7. Looks like problem is not with component, instead application should define focus order.
8. TBD
9. Confirmed in TEDI-READY
